files/servicenow.rb

```diff
@@ -85,15 +91,33 @@ def response
 end
 
 def servicenow(certname)
-  config = YAML.load_file('/etc/puppetlabs/puppet/servicenow.yaml')
-
-  instance          = config['instance']
-  username          = config['user']
-  password          = config['password']
-  table             = config['table']
-  certname_field    = config['certname_field']
-  classes_field     = config['classes_field']
-  environment_field = config['environment_field']
+  hiera_eyaml_config = nil
+  begin
+    # Note: If hiera-eyaml config doesn't exist, then load_config_file returns
+    # the hash {:options => {}, :sources => []}
+    hiera_eyaml_config = Hiera::Backend::Eyaml::Subcommand.load_config_file
+  rescue StandardError => e
+    raise "error reading the hiera-eyaml config: #{e}"
+  end
+
+  servicenow_config = File.read('/etc/puppetlabs/puppet/servicenow.yaml')
+  unless hiera_eyaml_config[:sources].empty?
+    # hiera-eyaml config exists so decrypt the servicenow_config. Note that
+    # the parser leaves plain-text values alone so this code should work OK
+    # even if the user specified plain-text config values.
+    Hiera::Backend::Eyaml::Options.set(hiera_eyaml_config[:options])
+    tokens = Hiera::Backend::Eyaml::Parser::ParserFactory.encrypted_parser.parse(servicenow_config)
+    servicenow_config = tokens.map(&:to_plain_text).join
+  end
+  servicenow_config = YAML.safe_load(servicenow_config)
+
+  instance          = servicenow_config['instance']
```

Comment:
Technically, this implementation means that all these fields could be `hiera-eyaml` encrypted values.